sdk/azidentity/managed_identity_client.go

```diff
@@ -191,6 +203,56 @@ func (c *managedIdentityClient) createAppServiceAuthRequest(ctx context.Context,
 	return request, nil
 }
 
+func (c *managedIdentityClient) getAzureArcSecretKey(ctx context.Context, scopes []string) (string, error) {
+	// create the request to retreive the secret key challenge provided by the HIMDS service
+	request, err := azcore.NewRequest(ctx, http.MethodGet, c.endpoint)
+	if err != nil {
+		return "", err
+	}
+	request.Header.Set(azcore.HeaderMetadata, "true")
+	q := request.URL.Query()
+	q.Add("api-version", azureArcAPIVersion)
+	q.Add("resource", strings.Join(scopes, " "))
```

Comment:
Is this using `scopes` for something other than authentication scopes, or does Arc call this `resource`?

Reply:
Arc calls it resource, also all the other Managed Identity credential scenarios call it resource and expect a resource. We have the code that trims the suffix in the AuthenticationPolicy code.

Reply:
Yes I remember now.  We're using `TokenRequestOptions.Scopes` for something other than scopes for MSI.

At the very least, the parameter names should be changed from `scopes` to `resources` as they aren't scopes.

Second, if this is how we intend to implement and publicly support this, we should document `TokenRequestOptions.Scopes` that for MSI it isn't a slice of scopes but a slice of resources.

Reply:
Which makes me wonder if we should rename `Scopes` to something more applicable, or just have a `Resources` field (I prefer the latter as it's explicit).

Also, does the suffix trimming in `AuthenticationPolicy()` work for all cases?  I recall that some scopes use "//.default".  The current algorithm will leave a trailing '/' in that case, is that ok/expected?  We had problems with confusion over this in track 1.

Reply:
I can rename the scopes params in MSI to `resources` but I think I'll do it in a separate PR to do it for all of the methods that call it scopes. Also, I can add a comment explaining that for MSI this field will contain resources instead of scopes. I can also fix the case where we have `//.default`.

Reply:
Be aware that some of them do have a trailing '/' while others do not.  See [environments](https://github.com/Azure/go-autorest/blob/master/autorest/azure/environments.go#L107) in `go-autorest` for examples (I assume this is the same thing).  Are the scopes uniform in this way?  This seems brittle.

Reply:
Hmmm that makes me think that removing only the `/.default` is on purpose. I can ask Scott about this, but it seems to me to account for the situation in which some still need to have a trailing `/`

Reply:
It could be.  Confirmation would be great (also update the algorithm's comment explaining this).

Reply:
Also, about my previous comment in AuthenticationPolicy we trim the suffix because we know we'll be most likely receiving a scope from the client lib.

Reply:
Ok sounds good, how about I make all of those improvements to the scopes vs resources in a separate PR though? I can rename scopes to resources for this specific method in this one though.